### PR TITLE
CV-Unity Location Settings

### PR DESCRIPTION
**Pull request by ah1053**
Location Settings
Code changes in [LocationSettings.cs](https://github.com/mobiledgex/edge-cloud-sampleapps/blob/cv-unity-location-settings/unity/ComputerVision-UnityClient/Assets/Scripts/LocationSettings.cs) , [LoadingIcon.cs](https://github.com/mobiledgex/edge-cloud-sampleapps/blob/cv-unity-location-settings/unity/ComputerVision-UnityClient/Assets/Scripts/LoadingIcon.cs), [NetworkManager.cs](https://github.com/mobiledgex/edge-cloud-sampleapps/blob/cv-unity-location-settings/unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs) .

Screen Shot 2020-09-18 at 3 11 59 PM

### CODE REVIEW

File: unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs [hunk 1]
```diff
@@ -49,17 +46,24 @@ public class NetworkManager:MonoBehaviour
...
             Rest,
             WebSocket
         }
         public ConnectionMode connectionMode;
+        public static int regionIndex = 0;
+
         #region MonoBehaviour Callbacks
         IEnumerator Start()
         {
+            regionsDmeList = new List<string>() { "wifi", "eu-mexdemo", "us-mexdemo", "jp-mexdemo", "kr-mexdemo" };
```
Comment (brucearmstrong) [inline unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs:64]: We maintain a list of DMEs at http://dme-inventory.mobiledgex.net/dme-list.html. If you use this list to populate your DME values, then we can easily add a new DME for someone without having to rebuilt the app. This is not an immediate requirement, but a "would be nice".

**Comment by brucearmstrong** [comment]
One improvement I see is for the "Use Device Location" to be a toggle button, that when enabled, disables the lat/lng fields. This way it is obvious that those values won't be used, even if they are shown in the UI, because they would be grayed out.

**Review by brucearmstrong — APPROVED** [review]
Good stuff. Very useful.

File: unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs (hunk 1 above)
Comment (lgarner) [reply unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs:64]: Shouldn't this list be pulled from the Unity SDK Editor? (who will start with a default list, and pull it from a server somewhere else)?

File: unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs [hunk 2]
```diff
@@ -239,7 +266,7 @@ void UpdateStats()
 
         public void UpdateAverageLatency(ConcurrentQueue<float> queue, float requestLatency)
         {
-            if (queue.Count < 10)
+            if (queue.Count < 100)
```
Comment (lgarner) [inline unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs:269]: Is 100 too high to see any changes in a reasonable amount of samples?

File: unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs (hunk 1 above)
Comment (ah1053) [reply unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs:64]: @lgarner  you have a point, will update in the next SDK release and then pull it in the app from the SDK.
@brucearmstrong  I was searching for something like that, Thanks.

File: unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs (hunk 2 above)
Comment (ah1053) [reply unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs:269]: @brucearmstrong  What is your take in that? I updated that based on the JS client & android app RollingAverageCalculation

Comment (ah1053) [reply unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs:269]: If I have full latency of 80 ms, then the average should change every 8 seconds

Comment (brucearmstrong) [reply unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs:269]: The 100 value in RollingAverage.java is the maximum size of a rolling window of data points. On sample #101, sample #1 is dropped.

**Pull request merged**